packages/block-library/src/gallery/block.json

```diff
@@ -68,14 +31,39 @@
 			"type": "boolean",
 			"default": true
 		},
+		"linkTarget": {
+			"type": "string"
+		},
 		"linkTo": {
 			"type": "string"
 		},
 		"sizeSlug": {
 			"type": "string",
 			"default": "large"
+		},
+		"allowResize": {
+			"type": "boolean",
+			"default": false
+		},
+		"isGrouped": {
+			"type": "boolean",
+			"default": true
+		},
+		"imageCount": {
+			"type": "number",
+			"default": 0
+		},
+		"gutterSize": {
```

Comment:
Do we need all these 5 new attributes here? Shouldn't we try to match the existing block without new features first?

Reply:
Sorry, forgot to respond to this one. We were thinking this was a low risk extra feature that may help with buy-in given the large breaking nature of the change, ie. at least brings one new feature. @paaljoachim had some thoughts about the value of including this additional feature in the first release.

Reply:
Oh I definitely, think it's a good feature to add. What I'm not sure about is the implementation itself. For instance, It's potentially something useful for columns/buttons/social links blocks as well. So I was thinking that if it's in its own PR, it would make it easier to reason about it more globally (maybe be a support flag...) in order for it to be supported by any horizontal container.

Reply:
A separate PR to consider more globally would normally make the most sense and I'd be happy to go with that if others agree. My primary concern with that is there could be a long time before the feature is available to users. 

Leveraging a block support flag is the first option that comes to mind. There is already work in the pipeline to add margins control to the spacing support. One problem here though is that the block style provided by that needs to be applied to inner elements for the gallery. There's been pushback on doing this sort of thing until it is more natively provided by the block support mechanism. An effort to provide that feature is under way in https://github.com/WordPress/gutenberg/issues/28913.

Another concern is that the current layout for the gallery relies on using the gutter size in width styles/calculations. For example: `width: calc(50% - (var(--gallery-block--gutter-size, #{$grid-unit-20}) / 2));`

The use of CSS variables for block supports has been avoided (rationale I was given https://github.com/WordPress/gutenberg/issues/22296) so if using a more global approach, I think we'll need to come up with a different layout solution. It would also need to still be easily handled by plugins that wish to provide other layouts through different means e.g. JS and masonry layouts.

Reply:
@aaronrobertshaw for the sake of getting this PR over the line as soon as possible maybe lets pull this out and aim for a follow up PR for this as soon as we get this one into master, we can then hopefully still get something included by the time the gallery refactor gets through call for testing and out of the experimental phase.

Reply:
I honestly understand that it's not great to receive pushback for things that actually work well and I'd approve in 99% of the cases where I'd be just "merge and iterate". I hate to give this kind of feedback personally.

The reality though is that the decisions we make on Gutenberg have bigger impacts than decisions we make on most projects I've worked on before personally and it all comes down to the backward compatibility strategy of WordPress. For these reasons, sometimes we need to be a bit more careful, hold back on some things until we understand the global impact. Things are definitely slower because of this, but that's just the approach in WordPress, take a bit more time to hopefully get it done as properly as we can (we do make mistakes regardless and it's fine)

Reply:
I agree, it's the right decision to remove the gutter size feature for now. No issues with the pushback at all 🙂 

We were looking to remove it once we'd solved one more outstanding issue with the PR in its current state.

Sometimes it's still worth asking the question and raising the concerns. It's easy to get conflicting advice on the right path forward. For example, we've been encouraged previously to use an adhoc approach if its a specialised need for an individual block. My thinking was the use of the gutter size in the width calculation may have qualified as such.

Thanks for all the great feedback on this PR!

Reply:
A great discussion here! Pulling out the gutter for the first iteration feels natural based on the really good discussion above here. I believe the main point is that gutter is a new feature and having a separate PR for it feels like the best way forward. 

This also means that one can at any time begin a PR for the gutter control to get the discussion going on how to implement the feature in the best possible way. A feature that can also become implemented in multiple blocks. I do look forward to seeing the gutter feature in the Gallery block.